beacon/lighthouse/subscription.go

```diff
@@ -0,0 +1,64 @@
+package lighthouse
+
+import (
+	"time"
+
+	"github.com/prometheus/common/log"
+
+	"github.com/alethio/eth2stats-client/types"
+)
+
+type ChainHeadSubscription struct {
+	data   chan types.ChainHead
+	client *LighthouseHTTPClient
+
+	stopChan chan bool
+}
+
+func NewChainHeadSubscription(client *LighthouseHTTPClient) *ChainHeadSubscription {
+	return &ChainHeadSubscription{
+		data:   make(chan types.ChainHead),
+		client: client,
+	}
+}
+
+func (s *ChainHeadSubscription) Start() {
+	log.Info("polling for new heads")
+	var lastHead *types.ChainHead
+
+	for {
+		select {
+		case <-s.stopChan:
+			close(s.data)
+			return
+		default:
+			head, err := s.client.GetChainHead()
+			if err != nil {
+				log.Errorf("failed to poll for chain head")
+				close(s.data)
+				return
+			}
+			if lastHead == nil || *lastHead != *head {
+				s.data <- types.ChainHead{
+					HeadSlot:           head.HeadSlot,
+					HeadBlockRoot:      head.HeadBlockRoot,
+					FinalizedSlot:      head.FinalizedSlot,
+					FinalizedBlockRoot: head.FinalizedBlockRoot,
+					JustifiedSlot:      head.JustifiedSlot,
+					JustifiedBlockRoot: head.JustifiedBlockRoot,
+				}
+				lastHead = head
+			}
+
+			time.Sleep(time.Second)
```

Comment:
It might be a better idea to extract this into a constant

Reply:
done, was just being lazy